code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm

```diff
@@ -359,4 +359,407 @@ Difficulty: Very Hard
+	cooldown_add = 200
+	var/terrain_theme = "winter"
+	var/NewTerrainFloors
+	var/NewTerrainWalls
+	var/NewTerrainChairs
+	var/NewTerrainTables
+	var/list/NewFlora = list()
+	var/florachance = 8
+	var/FloorLayerChange
+
+/obj/machinery/anomalous_crystal/theme_warp/New()
+	..()
+	terrain_theme = pick("lavaland","winter","jungle","ayy lmao")
+	switch(terrain_theme)
+		if("lavaland")//Depressurizes the place... and free cult metal, I guess.
+			NewTerrainFloors = /turf/open/floor/plating/asteroid/basalt/lava_land_surface
+			NewTerrainWalls = /turf/closed/wall/mineral/cult
+			NewFlora = list(/mob/living/simple_animal/hostile/asteroid/goldgrub)
+			florachance = 1
+			FloorLayerChange = 2.6
+		if("winter") //Snow terrain is slow to move in and cold! Get the assistants to shovel your driveway.
+			NewTerrainFloors = /turf/open/floor/plating/asteroid/snow/atmosphere
+			NewTerrainWalls = /turf/closed/wall/mineral/wood
+			NewTerrainChairs = /obj/structure/chair/wood/normal
+			NewTerrainTables = /obj/structure/table/glass
+			NewFlora = list(/obj/structure/flora/grass/green, /obj/structure/flora/grass/brown, /obj/structure/flora/grass/both)
+			FloorLayerChange = 2.6
+		if("jungle") //Beneficial due to actually having breathable air. Plus, monkeys.
+			NewTerrainFloors = /turf/open/floor/grass
+			NewTerrainWalls = /turf/closed/wall/mineral/sandstone
+			NewTerrainChairs = /obj/structure/chair/wood
+			NewTerrainTables = /obj/structure/table/wood
+			NewFlora = list(/obj/structure/flora/ausbushes/sparsegrass, /obj/structure/flora/ausbushes/fernybush, /obj/structure/flora/ausbushes/leafybush,
+							/obj/structure/flora/ausbushes/grassybush, /obj/structure/flora/ausbushes/sunnybush, /obj/structure/flora/tree/palm, /mob/living/carbon/monkey)
+			florachance = 20
+			FloorLayerChange = 2.6
+		if("ayy lmao") //Beneficial, turns stuff into alien alloy which is useful to cargo and research. Also repairs atmos.
+			NewTerrainFloors = /turf/open/floor/plating/abductor
+			NewTerrainWalls = /turf/closed/wall/mineral/abductor
+			NewTerrainChairs = /obj/structure/bed/abductor //ayys apparently don't have chairs. An entire species of people who only recline.
+			NewTerrainTables = /obj/structure/table/abductor
+
+/obj/machinery/anomalous_crystal/theme_warp/ActivationReaction(mob/user, method)
+	if(..())
+		var/area/A = get_area(src)
+		if(!A.outdoors && !(A in affected_targets))
+			for(var/atom/Stuff in A)
+				if(isturf(Stuff))
+					var/turf/T = Stuff
+					if((istype(T, /turf/open/space) || istype(T, /turf/open/floor)) && NewTerrainFloors)
+						var/turf/open/O = T.ChangeTurf(NewTerrainFloors)
+						if(O.air)
+							var/datum/gas_mixture/G = O.air
+							G.copy_from_turf(O)
+						if(FloorLayerChange)
+							O.layer = FloorLayerChange
+						if(prob(florachance) && NewFlora.len)
+							var/density_checks = 1
+							for(var/atom/junk in O)
+								if(junk.density)
+									density_checks = 0
+									break
+							if(density_checks)
```

Comment:
Instead of all those lines, just do `if(!is_blocked_turf(O))` on this one for density checks